### PR TITLE
Support for custom UI and video component

### PR DESCRIPTION
**Pull request by hans00**
Add support for custom UI.

And I implemented the video view component (embedded the Zoom conference video, android only).
But it seems conflict with RN Animated(blank view), could I contribute these code?

### CODE REVIEW

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 1]
```diff
@@ -281,6 +332,287 @@ public void connectAudio() {
     connectAudioWithVoIP();
   }
 
+  @ReactMethod
+  public void isMeetingConnected(Promise promise) {
+    WritableArray rnUserList = Arguments.createArray();
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:337]: unused
```suggestion
```

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 2]
```diff
@@ -281,6 +332,287 @@ public void connectAudio() {
...
+    }
+
+    final InMeetingService inMeetingService = zoomSDK.getInMeetingService();
+
+    promise.resolve(inMeetingService.isMeetingConnected());
+  }
+
+  @ReactMethod
+  public void isMeetingHost(Promise promise) {
+    WritableArray rnUserList = Arguments.createArray();
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:352]: unused
```suggestion
```

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 3]
```diff
@@ -281,6 +332,287 @@ public void connectAudio() {
...
+      return;
+    }
+
+    final InMeetingService inMeetingService = zoomSDK.getInMeetingService();
+
+    promise.resolve(inMeetingService.isMeetingHost());
+  }
+
+  @ReactMethod
+  public void getInMeetingUserList(Promise promise) {
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:366]: you are fetching user ids, not users
```suggestion
  public void getInMeetingUserIdList(Promise promise) {
```

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 4]
```diff
@@ -281,6 +332,287 @@ public void connectAudio() {
...
+    final InMeetingService inMeetingService = zoomSDK.getInMeetingService();
+
+    final InMeetingAudioController audioController = inMeetingService.getInMeetingAudioController();
+
+    MobileRTCSDKError result = audioController.muteMyAudio(muted);
+
+    if (result == MobileRTCSDKError.SDKERR_SUCCESS) {
+      promise.resolve(null);
+    } else {
+      promise.reject("ERR_ZOOM_MEETING", "Mute myself error, status: " + result.name());
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:443]: to be in sync with `Mute my video`
```suggestion
      promise.reject("ERR_ZOOM_MEETING", "Mute my audio error, status: " + result.name());
```

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 5]
```diff
@@ -281,6 +332,287 @@ public void connectAudio() {
...
+    final InMeetingService inMeetingService = zoomSDK.getInMeetingService();
+
+    final InMeetingAudioController audioController = inMeetingService.getInMeetingAudioController();
+
+    MobileRTCSDKError result = audioController.muteAttendeeAudio(muted, Long.parseLong(userId));
+
+    if (result == MobileRTCSDKError.SDKERR_SUCCESS) {
+      promise.resolve(null);
+    } else {
+      promise.reject("ERR_ZOOM_MEETING", "Mute error, status: " + result.name());
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:464]:
```suggestion
      promise.reject("ERR_ZOOM_MEETING", "Mute attendee audio error, status: " + result.name());
```

**Review by mieszko4 — APPROVED** [review]
Thnx @hans00 !
I see you've put a lot of effort into this pr.
I left you few suggestions.
If it passes the smoke test (i.e. it is not breaking any of the existing functionality), let's merge it!

Could you confirm it passes the smoke test?
https://github.com/mieszko4/react-native-zoom-us#testing

File: ios/RNZoomUs.m [hunk 6]
```diff
@@ -215,6 +215,200 @@ - (void)connectAudio {
...
+
+RCT_EXPORT_METHOD(stopShareScreen: (BOOL)muted resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  @try {
+    MobileRTCMeetingService *ms = [[MobileRTC sharedRTC] getMeetingService];
+    if (ms) {
+      [ms stopAppShare];
+    }
+    resolve(nil);
+  } @catch (NSError *ex) {
+    reject(@"ERR_UNEXPECTED_EXCEPTION", @"Executing stopShareScreen", ex);
```
Comment (mieszko4) [inline ios/RNZoomUs.m:359]: I've noticed that some rejections are not consistent between ios and android
android: `promise.reject("ERR_ZOOM_MEETING_CONTROL", "Stop share screen error, status: " + result.name());`

File: README.md [hunk 7]
```diff
@@ -47,6 +47,16 @@ buildscript {
 
 See [diff](https://github.com/mieszko4/react-native-zoom-us-test/pull/10/commits/cabdb876cc40f78f0a6d977d38377497be5e0726) for reference.
 
+3. Optional: Add custom activity config (`android/app/src/main/res/values/config.xml`)
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="zm_config_conf_activity">ID of your custom activity</string>
+</resources>
```
Comment (mieszko4) [inline README.md:55]: It says `Optional` - but it is not clear at which situations this should be set

**Review by mieszko4 — APPROVED** [review]
:cool:

**Comment by hans00** [comment]
Ummm... In my work, currently not need iOS.
And the VideoView for iOS, I'm not sure how to implement. It seems have different behavior (not single component).
So, I just implement in-meeting control.

**Comment by hans00** [comment]
Almost function are work fine.
But some method have weird bug.

- `raiseHand` / `lowerHand`: throw the `SDK_ERROR` (SDK bug?
- `switchCamera`: On my test device, will break preview of VideoView and switch back will black on others (SDK bug or device compatibility?
- Re add VideoView with preview after `switchCamera`, the camera will back to first camera (SDK bug or device compatibility?

**Review by mieszko4 — APPROVED** [review]
looks good.
Should we merge it?

**Comment by hans00** [comment]
It seems done
You can merge it

**Pull request merged**